### PR TITLE
Add TypeScript support, config and update dependencies

### LINKED ISSUE #65: Migrate to TypeScript
## Problem

1. This tool was built in JavaScript to be a simplest example of Editor.js Tools API usage. Right now further enhancements requires more modern stack.
2. When using in TypeScript projects there is a problem with lack of type declarations. See https://github.com/codex-team/editor.js/issues/2143

## Task

We need to migrate this project to TypeScript and provide type declarations

### PR DESCRIPTION
**Pull request by aavishkarmishra**
Resolve #65   Migrate to Typescript

Changes:

Add Typescript
Add type dependencies @editorjs/editorjs
Add tsconfig.json
Convert Js files to Ts files
Add .d.ts for @codexteam/icons
Add vite-plugin-dts to generate types during build

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+  caption: string;
+}
+
+/**
+ * @class Quote
+ * @classdesc Quote Tool for Editor.js
+ * @property {QuoteData} data - Tool`s input and output data
+ * @propert {API} api - Editor.js API instance
+ */
+export default class Quote {
```
Comment (neSpecc) [inline src/index.ts]: it should implement `BlockTool` interface from `@editorjs/editorjs`

File: package.json [hunk 2]
```diff
@@ -21,19 +21,24 @@
       "require": "./dist/quote.umd.js"
     }
   },
+  "types": "./dist/index.d.ts",
```
Comment (neSpecc) [inline package.json:25]: types also should be declared under "exports", take look here: https://github.com/editor-js/nested-list/pull/73/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R23

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@editorjs/quote",
-  "version": "2.6.0",
+  "version": "2.6.1",
```
Comment (neSpecc) [inline package.json]:
```suggestion
  "version": "2.7.0",
```

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+
+/**
+ * @typedef {object} TunesMenuConfig
+ * @property {string} icon - menu item icon
+ * @property {string} label - menu item label
+ * @property {boolean} isActive - true if item should be in active state
+ * @property {boolean} closeOnActivate - if true tunes menu should close once any item is selected
+ * @property {() => void} onActivate - item activation callback
+ */
+interface TunesMenuConfig {
```
Comment (neSpecc) [inline src/index.ts]: you can use `MenuConfig` type

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+    return "Enter a caption";
+  }
+
+  /**
+   * Allowed quote alignments
+   *
+   * @public
+   * @returns {{left: string, center: string}}
+   */
+  static get ALIGNMENTS(): { left: "left"; center: "center" } {
```
Comment (neSpecc) [inline src/index.ts]: I'd suggest to create Enum `Alignment` with `Left` and `Center`

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,477 @@
...
+
+  /**
+   * Helper for making Elements with attributes
+   *
+   * @param  {string} tagName           - new Element tag name
+   * @param  {Array|string} classNames  - list or name of CSS classname(s)
+   * @param  {object} attributes        - any attributes
+   * @returns {Element}
+   */
+  _make(
```
Comment (neSpecc) [inline src/index.ts]: lets use [make](https://github.com/editor-js/utils/blob/main/packages/dom/src/make.ts) from `@editorjs/dom`

```
npm install @editorjs/dom
```

File: src/types/icons.d.ts
Comment (neSpecc) [inline src/types/icons.d.ts:1]: update `@codexteam/icons` to 0.3.2 where types should work fine

Comment (aavishkarmishra) [reply src/types/icons.d.ts:1]: done👍🏼

File: src/index.ts (hunk 6 above)
Comment (aavishkarmishra) [reply src/index.ts]: done👍🏼

File: src/index.ts (hunk 5 above)
Comment (aavishkarmishra) [reply src/index.ts]: done👍🏼

File: src/index.ts (hunk 4 above)
Comment (aavishkarmishra) [reply src/index.ts]: done👍🏼

File: package.json (hunk 3 above)
Comment (aavishkarmishra) [reply package.json]: done👍🏼

File: package.json (hunk 2 above)
Comment (aavishkarmishra) [reply package.json:25]: done👍🏼

File: src/index.ts (hunk 1 above)
Comment (aavishkarmishra) [reply src/index.ts]: done👍🏼

File: package.json [hunk 7]
```diff
@@ -36,9 +37,10 @@
     "typescript": "^5.4.5",
     "vite": "^4.5.0",
     "vite-plugin-css-injected-by-js": "^3.3.0",
-    "vite-plugin-dts": "^3.9.1"
+    "vite-plugin-dts": "^3.9.1",
+    "@editorjs/dom": "^0.0.1"
```
Comment (neSpecc) [inline package.json]: its a prod dependency, since `make` is required for tool to work

File: src/index.ts [hunk 8]
```diff
@@ -145,13 +112,22 @@ interface QuoteCSS {
   caption: string;
 }
 
+/**
+ * @typedef {Enum} Alignment
+ * @description Enum for Quote Alignment
+ */
+enum Alignment {
+  left = "left",
```
Comment (neSpecc) [inline src/index.ts]: its better to [capitalize](https://www.typescriptlang.org/docs/handbook/enums.html) enum constants

File: package.json (hunk 7 above)
Comment (aavishkarmishra) [reply package.json]: Done 👍🏼

File: src/index.ts (hunk 8 above)
Comment (aavishkarmishra) [reply src/index.ts]: Done 👍🏼

Comment (neSpecc) [reply src/index.ts]: I mean `Alignment.Left` and `Alignment.Right` like in the official linked above

File: src/index.ts [hunk 9]
```diff
@@ -0,0 +1,426 @@
...
+    return "Enter a caption";
+  }
+
+  /**
+   * Allowed quote alignments
+   *
+   * @public
+   * @returns {Record<string, ALIGNMENT> }
+   */
+  static get ALIGNMENTS(): Record<string, ALIGNMENT> {
```
Comment (neSpecc) [inline src/index.ts]: seem redundant since we have enum

Comment (aavishkarmishra) [reply src/index.ts]: Done 👍🏼

**Review by neSpecc — APPROVED** [review]

**Pull request merged**